wagtail/wagtailsearch/tests/test_frontend.py

```diff
@@ -1,25 +1,120 @@
+        self.assertEqual(list(search_results), [christmas_event_page])
+
+        # Check the query object
+        self.assertIsInstance(response.context['query'], Query)
+        query = response.context['query']
+        self.assertEqual(query.query_string, "christmas")
+
+    def pagination_test(test):
+        def wrapper(*args, **kwargs):
+            # Create some pages
+            event_index = Page.objects.get(url_path='/home/events/')
+            for i in range(100):
+                event = EventPage(
+                    title="Event " + str(i),
+                    slug='event-' + str(i),
+                    live=True,
+                )
+                event_index.add_child(instance=event)
+
+            return test(*args, **kwargs)
+
+        return wrapper
+
+    @pagination_test
+    def test_get_first_page(self):
+        response = self.client.get(reverse('wagtailsearch_search') + '?q=Event&page=1')
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'wagtailsearch/search_results.html')
+
+        # Test that we got the first page
+        search_results = response.context['search_results']
+        self.assertEqual(search_results.number, 1)
+
+    @pagination_test
+    def test_get_10th_page(self):
+        response = self.client.get(reverse('wagtailsearch_search') + '?q=Event&page=10')
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'wagtailsearch/search_results.html')
+
+        # Test that we got the tenth page
+        search_results = response.context['search_results']
+        self.assertEqual(search_results.number, 10)
+
+    @pagination_test
+    def test_get_invalid_page(self):
+        response = self.client.get(reverse('wagtailsearch_search') + '?q=Event&page=Not a Page')
         self.assertEqual(response.status_code, 200)
-        self.assertEqual(response.context['query_string'], "Hello")
+        self.assertTemplateUsed(response, 'wagtailsearch/search_results.html')
+
+        # Test that we got the first page
+        search_results = response.context['search_results']
+        self.assertEqual(search_results.number, 1)
+
+    @pagination_test
+    def test_get_out_of_range_page(self):
+        response = self.client.get(reverse('wagtailsearch_search') + '?q=Event&page=9999')
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'wagtailsearch/search_results.html')
+
+        # Test that we got the last page
+        search_results = response.context['search_results']
+        self.assertEqual(search_results.number, search_results.paginator.num_pages)
+
+    @pagination_test
+    def test_get_zero_page(self):
+        response = self.client.get(reverse('wagtailsearch_search') + '?q=Event&page=0')
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'wagtailsearch/search_results.html')
+
+        # Test that we got the first page
```

Comment:
Misleading comment here. I meant to say "Test that we got the last page".

I think it makes sense that if a user just happened to stumble across page 0 they should be given page 1. But since this is very unlikely to happen, I decided to make this behave like the rest of Wagtail and get the last page (and forgot to change the comment!).